builtin/mainmenu/content/pkgmgr.lua

```diff
@@ -711,6 +711,53 @@ function pkgmgr.preparemodlist(data)
 	return retval
 end
 
+function pkgmgr.prepareclientmodlist(data)
+	local retval = {}
+
+	local clientmods = {}
+
+	--read clientmods
+	local modpath = core.get_clientmodpath()
+
+	if modpath ~= nil and modpath ~= "" then
+		pkgmgr.get_mods(modpath, "clientmods", clientmods)
+	end
+
+	for i=1,#clientmods,1 do
+		clientmods[i].type = "mod"
+		clientmods[i].loc = "global"
+		clientmods[i].is_clientside = true
+		retval[#retval + 1] = clientmods[i]
+	end
+
+	--read mods configuration
+	local filename = modpath ..
+				DIR_DELIM .. "mods.conf"
+
+	local conffile = Settings(filename)
+
+	for key,value in pairs(conffile:to_table()) do
+		if key:sub(1, 9) == "load_mod_" then
+			key = key:sub(10)
+			local element = nil
+			for i=1,#retval,1 do
```

Comment:
Bad style. Just do something like

```suggestion
			for _, client_mod in ipairs(client_mods) do
```

and give `element` a better name (such as `found_client_mod` or whatever).

though really i feel you should have client mods indexed in a table by name. then you wouldn't need a linear search here - the code would be cleaner.